src/unix/linux/system.rs

```diff
@@ -385,23 +385,43 @@ impl SystemInner {
         get_system_info_android(InfoType::Name)
     }
 
+    #[cfg(not(target_os = "android"))]
     pub(crate) fn long_os_version() -> Option<String> {
-        #[cfg(target_os = "android")]
-        let system_name = "Android";
+        let mut long_name = "Linux".to_owned();
+
+        let distro_name = Self::name();
+        let distro_version = Self::os_version();
+        if let (Some(distro_name), Some(distro_version)) = (&distro_name, &distro_version) {
+            // "Linux (Ubuntu 24.04)"
+            long_name.push_str(" (");
+            long_name.push_str(distro_name);
+            long_name.push(' ');
+            long_name.push_str(distro_version);
+            long_name.push(')');
+        } else if let Some(distro_or_version) = distro_name.or(distro_version) {
+            long_name.push_str(" (");
+            long_name.push_str(&distro_or_version);
+            long_name.push(')');
```

Comment:
This part might be weird &mdash; I am not sure what circumstances would lead to `System::name()` being Some and `System::os_version()` being None, or vice versa. The former wouldn't be an issue: "Linux (Ubuntu)" is perfectly fine. The latter is worse: "Linux (24.04)".

I considered leaving out os_version() if name() is None, but then we have a situation that the value of long_os_version() specifically does not include the os_version().

Reply:
I think adding "unknown" in case the OS name is not unknown would likely improve the situation? If the version is unknown then just "Linux (Ubuntu)" sounds fine I think.

Reply:
Yes, good call. I like that.

Speaking of "unknown", I forgot about checking for "unknown" in the Android case which is something I mentioned in https://github.com/GuillaumeGomez/sysinfo/issues/1419#issuecomment-2526607322. Because `getString` in Build.java can return "unknown":

- https://android.googlesource.com/platform/frameworks/base/+/bc3378bf08f908335c721e597f51f6b95374f0c8/core/java/android/os/Build.java#137
- https://android.googlesource.com/platform/frameworks/base/+/bc3378bf08f908335c721e597f51f6b95374f0c8/core/java/android/os/Build.java#1564
- https://android.googlesource.com/platform/frameworks/base/+/bc3378bf08f908335c721e597f51f6b95374f0c8/core/java/android/os/Build.java#55

But looking at this more, I think "unknown" is specific to the Java API. They have `public static final String MODEL` where they have chosen to use the string "unknown" instead of `null` when a particular property hasn't been set. Whereas in Rust, we deal with `libc::__system_property_get` returning 0 by turning it into None (`System::name()` returns `Option<String>` not `String`). So I think we do not need to check for "unknown" after all.